src/bun.zig

```diff
@@ -3928,3 +3928,34 @@ pub fn indexOfPointerInSlice(comptime T: type, slice: []const T, item: *const T)
     const index = @divExact(offset, @sizeOf(T));
     return index;
 }
+
+pub fn getThreadCount() u16 {
+    const max_threads = 1024;
+    const min_threads = 2;
+    const ThreadCount = struct {
+        pub var cached_thread_count: u16 = 0;
+        var cached_thread_count_once = std.once(getThreadCountOnce);
+        fn getThreadCountFromUser() ?u16 {
+            inline for (.{ "UV_THREADPOOL_SIZE", "GOMAXPROCS" }) |envname| {
+                if (getenvZ(envname)) |env| {
+                    if (std.fmt.parseInt(u16, env, 10) catch null) |parsed| {
+                        if (parsed >= min_threads) {
```

Comment:
GOMAXPROCS=1 is a thing right? Should min_threads be 1 instead of 2?

Reply:
it should be 2

Reply:
sometimes we depend on threads to exist